x-pack/plugins/reporting/server/lib/event_logger/logger.ts

```diff
@@ -27,9 +28,39 @@ export interface ExecutionCompleteMetrics {
   byteSize: number;
 }
 
+const getEventLog = (logger: LevelLogger) => ({
+  getLogger(lProperties: Partial<IEvent>): IEventLogger {
+    return (function () {
+      let start: Date | undefined;
+      let end: Date | undefined;
+      let duration: number | undefined;
+
+      return {
+        logEvent(properties: IEvent) {
+          if (start && !end) {
+            end = start ? new Date() : undefined;
+          }
+          duration = end && start ? end.valueOf() - start.valueOf() : undefined;
+          const timing: Partial<IEvent> = {
+            event: { duration, start: start?.toISOString(), end: end?.toISOString() },
+          };
+          const mProperties: Partial<IEvent> = deepMerge(lProperties, timing);
+          logger.debug(JSON.stringify(deepMerge(mProperties, properties)), ['reporting-event-log']);
```

Comment:
I think we should consider making use the second, optional param to the logger methods, which is basically an ECS record.  I suspect the fields in the JSONified string are going to be hard to search for, as long as the JSON layout is used for the Kibana logger, it should write the **standard** (not custom though) fields out.

That might mean changing the string message parameters on these bit, to make them more useful to humans.

I poked around and found this PR which brought some of the "ECS-ish" bits to the Kibana log, which is about the total of my knowledge on this subject.  Not clear to me if this would be more useful than the JSONified string, but guessing it could be, at least in some cases ...  https://github.com/elastic/kibana/pull/96350

Reply:
Thanks so much for this insight, @pmuellr! I followed your advice and updated the debug log entries to log a simple message (rather than stringified JSON), and use the LogMega param: 456079c10ba